src/resources/Catalogs/CatalogInterfaces.ts

```diff
@@ -65,7 +65,7 @@ export interface CatalogModel extends BaseCatalogModel {
     /**
      * @deprecated use `configuration.product` instead.
      */
-    product: ProductHierarchyModel;
+    product?: ProductHierarchyModel;
```

Comment:
Shouldn't only the `product` property in the `CreateCatalogModel` be optional and this one left required as it's only when creating a catalog that we don't require a `product` property if the configuration is provided?